druntime/src/rt/lifetime.d

```diff
@@ -1758,84 +1758,71 @@ do
 Given an array of length `size` that needs to be expanded to `newlength`,
 compute a new capacity.
 
-Better version by Dave Fladebo:
+Better version by Dave Fladebo, enhanced by Steven Schveighoffer:
 This uses an inverse logorithmic algorithm to pre-allocate a bit more
 space for larger arrays.
-- Arrays smaller than PAGESIZE bytes are left as-is, so for the most
-common cases, memory allocation is 1 to 1. The small overhead added
-doesn't affect small array perf. (it's virtually the same as
-current).
-- Larger arrays have some space pre-allocated.
+- The maximum "extra" space is about 80% of the requested space. This is for
+PAGE size and smaller.
 - As the arrays grow, the relative pre-allocated space shrinks.
-- The logorithmic algorithm allocates relatively more space for
-mid-size arrays, making it very fast for medium arrays (for
-mid-to-large arrays, this turns out to be quite a bit faster than the
-equivalent realloc() code in C, on Linux at least. Small arrays are
-just as fast as GCC).
 - Perhaps most importantly, overall memory usage and stress on the GC
 is decreased significantly for demanding environments.
+- The algorithm is tuned to avoid any division at runtime.
 
 Params:
     newlength = new `.length`
-    size = old `.length`
+    elemsize = size of the element in the new array
 Returns: new capacity for array
 */
-size_t newCapacity(size_t newlength, size_t size)
+size_t newCapacity(size_t newlength, size_t elemsize)
 {
     version (none)
     {
-        size_t newcap = newlength * size;
+        size_t newcap = newlength * elemsize;
     }
     else
```

Comment:
what is this `version(none)` doing here, and why keep it around?

Reply:
No idea. I imagine it's legacy from pre-D2.

Edit: yep

https://github.com/dlang/phobos/commit/736794d4af5c2b33267254c1b4261c7d56695d2d#diff-3a01d989234b395d8f6098c4849117d0d6135f340f1599849de03080ae7607bcR413

When the moon of mars was still cooling, this was there.

Reply:
I think I can remove it. There is zero chance it will ever get used.

Reply:
please do.

Reply:
Removed. I made it a separate commit to avoid a whitespace diff causing problems.